src/framework/control.ml

```diff
@@ -586,23 +582,21 @@ struct
 
     if get_bool "dbg.verbose" && get_string "result" <> "none" then print_endline ("Generating output: " ^ get_string "result");
     Result.output (lazy local_xml) gh make_global_fast_xml file
-
-
-  let analyze file fs change_info =
-    analyze file fs (get_spec ()) change_info
-
-  let rec analyze_loop file fs change_info =
-    try
-      analyze file fs change_info
-    with Refinement.RestartAnalysis ->
-      (* Tail-recursively restart the analysis again, when requested.
-         All solving starts from scratch.
-         Whoever raised the exception should've modified some global state
-         to do a more precise analysis next time. *)
-      (* TODO: do some more incremental refinement and reuse parts of solution *)
-      analyze_loop file fs change_info
 end
 
+let rec analyze_loop (module CFG : CfgBidir) file fs change_info =
```

Comment:
I think there should be a comment here about why `analyze_loop` cannot be part of `AnalyzeCFG` any more. Took me a while to figure out it's simply because each call of `get_spec` in the loop should return a different `Spec` module, so obviously the functor argument cannot change within the functor itself.

Reply:
Good point. I remember being confused by `analyze_loop` myself because I thought `get_spec` would always have the same result.

Reply:
It's a quite obscure and hacky thing anyway. I only added this mechanism for refining with witnesses, where the `ana.activated` option is modified before restarting, causing `MCP2` inside `get_spec ()` to be extended.